### PR TITLE
First pass at merges doc

### LINKED ISSUE #632: New Doc: SQL Reference/Version Control Features/Merge
1. How conflicts are calculated
    1. data
    1. schema
1. The `dolt_conflicts` and `dolt_conflicts_<tablename>` tables
    * Schema conflicts are not resolvable via SQL
3. Resolving conflicts in SQL 
    * Note must resolve in a single transaction
1. Persisting conflicts to resolve later (not default)

### PR DESCRIPTION
**Pull request by timsehn**
Fixes #632

### CODE REVIEW

File: content/reference/sql/version-control/merges.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+title: Merges
+---
+
+# Merging branches
+
+To merge a branch into your current branch, use the [`DOLT_MERGE()`
+stored procedure](dolt-sql-procedures.md#dolt_merge):
+
+```sql
+call dolt_merge('feature-branch');
```
Comment (druvv) [inline content/reference/sql/version-control/merges.md]: Nitpicky, but for docs format capitalization is nice:
```suggestion
CALL DOLT_MERGE('feature-branch');
```

File: content/reference/sql/version-control/merges.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+```
+
+This function returns whether the merge was a fast-forward or not and whether or not there are conflicts generated by the merge.
```
Comment (druvv) [inline content/reference/sql/version-control/merges.md:14]: the conflicts column is also `1` when there are constraint violations:

```suggestion
This function returns whether the merge was a fast-forward or not and whether or not there are conflicts/constraint violations generated by the merge.
```

Comment (druvv) [reply content/reference/sql/version-control/merges.md:14]: You could clarify that here, or in a separate doc?

Comment (druvv) [reply content/reference/sql/version-control/merges.md:14]: We could also add a new column called `constraint_violations`

File: content/reference/sql/version-control/merges.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+times when you need to commit merge conflicts, for example to
+collaborate on resolving them with other people. To allow committing
+merge conflicts, change this system variable to `1` for every client
+that needs to commit merge conflicts:
+
+```sql
+set @@dolt_allow_commit_conflicts = 1;
+```
+
+The server will not allow you to create new Dolt commits (with the
```
Comment (druvv) [inline content/reference/sql/version-control/merges.md:168]: Something that may be important to clarify is that the moment you sql transaction commit, commit conflicts into a working set, any other MySQL client has to set `@@dolt_allow_commit_conflicts = 1` even if they aren't producing new conflicts.

**Review by druvv — APPROVED** [review]
lgtm, what do you want to do about constraint violations?

File: content/reference/sql/version-control/merges.md (hunk 2 above)
Comment (timsehn) [reply content/reference/sql/version-control/merges.md:14]: I think we should.

File: content/reference/sql/version-control/merges.md (hunk 3 above)
Comment (timsehn) [reply content/reference/sql/version-control/merges.md:168]: what happens if they don't?

Comment (timsehn) [reply content/reference/sql/version-control/merges.md:168]: Feel free to add this note yourself.

File: content/reference/sql/version-control/merges.md (hunk 2 above)
Comment (timsehn) [reply content/reference/sql/version-control/merges.md:14]: You can fix this later.

**Pull request merged**